packages/opentelemetry-exporter-jaeger/src/jaeger.ts

```diff
@@ -55,7 +55,15 @@ export class JaegerExporter implements SpanExporter {
 
   /** Shutdown exporter. */
   shutdown(): void {
-    this._sender.close();
+    this._sender.flush((numSpans: number, err?: string) => {
+      if (err) {
+        this._logger.error(`failed to flush span: ${err}`);
+      }
+    });
+    // Sleeping 2 seconds before closing the sender's connection to ensure all spans are flushed.
+    setTimeout(() => {
+      this._sender.close();
+    }, 2000);
```

Comment:
I feel like we should add some configurable timeout for shutdown so that users can either give more time to flush span data or prevent it from blocking the app's shutdown.

Reply:
Created issue https://github.com/open-telemetry/opentelemetry-js/issues/284, will merge this one and work on that.